src/boimmgpy/model_annotation/model_annotator.py

```diff
@@ -0,0 +1,382 @@
+                + "' return ID(s) as boimmg_id"
+            )
+            data = result.data()
+            for value in data:
+                sidechain_id.append(value.get("boimmg_id"))
+
+        if len(backbone_id) != 0 and len(sidechain_id) != 0:
+            return backbone_id, sidechain_id, compound
+
+        if len(backbone_id) == 0 and len(sidechain_id) != 0:
+            result = session.run(
+                "match (c:Compound) where c.name='"
+                + str(backbone)
+                + "' return ID(c) as boimmg_id"
+            )
+            data = result.data()
+            for value in data:
+                backbone_id.append(value.get("boimmg_id"))
+            if backbone != 0:
+                compound = True
+                return backbone_id, sidechain_id, compound
+
+    @staticmethod
+    def get_coumpound(session,backbone: str, sidechains: list, flag: bool):
+        """Method that search for the the specific compounds attached for the side chain and backbone synonym's.
+        In case that the ID from the backbone is already from a compound it will search only for the side chains compounds.
+
+        Args:
+            backbone (str): Backbone's ID
+            sidechains (list): List of side chains ID's
+            flag (bool): Bolean is True when the backbone ID is from a compound
+
+        Returns:
+            _type_: Tuple with backbone and side chains Compounds ID's
+        """
+        side_chains_coumpound = []
+        backbone_coumpound = 0
+
+        if flag == False:
+            for v in sidechains:
+                result = session.run(
+                    "match(s:Synonym)-[:is_synonym_of]->(l)-[:is_db_link_of]->(t:Compound) where id(s)=$v and exists(t.boimmg_id) return id(t)",
+                    v=v,
+                )
+                data = result.data()
+                for value in data:
+                    if value.get("id(t)") not in side_chains_coumpound:
+                        side_chains_coumpound.append(value.get("id(t)"))
+            node = session.run(
+                "match(s:Synonym)-[:is_synonym_of]->(l)-[:is_db_link_of]->(t:Compound) where id(s)=$backbone and exists(t.boimmg_id) return t.boimmg_id as boimmg_id",
+                backbone=backbone,
+            )
+            for value in node:
+                backbone_coumpound = value.get("boimmg_id")
+            if len(side_chains_coumpound) != 0:
+                return backbone_coumpound, side_chains_coumpound
+
+        if flag == True:
+            for v in sidechains:
+                result = session.run(
+                    "match(s:Synonym)-[:is_synonym_of]->(l)-[:is_db_link_of]->(t:Compound) where id(s)=$v and exists(t.boimmg_id) return id(t)",
+                    v=v,
+                )
+                data = result.data()
+                for value in data:
+                    side_chains_coumpound.append(value.get("id(t)"))
+                return backbone, side_chains_coumpound
```

Comment:
This chunk of code is the same as above, wrap it into one method to avoid code repetition